packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -405,14 +406,12 @@ impl ScriptTransactionBuilder {
             .collect()
     }
 
-    fn no_spendable_input<'a, I: IntoIterator<Item = &'a FuelInput>>(inputs: I) -> bool {
+    fn no_input_to_cover_fees<'a, I: IntoIterator<Item = &'a FuelInput>>(inputs: I) -> bool {
         !inputs.into_iter().any(|i| {
             matches!(
                 i,
-                FuelInput::CoinSigned(_)
```

Comment:
Why are messages disregarded?

Reply:
They should be included. Pushed a new commit.